graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +39,340 @@ def __init__(self, query_ast):
         # List[SubQueryNode], the queries that depend on the current query
 
 
+def _split_query_ast_one_level_recursive(
+    query_node, ast, parent_selections, type_info, edge_to_stitch_fields,
+    intermediate_out_name_assigner
+):
+    """Return a Node to replace the input AST by in the selections one level above.
+
+    - If the input AST starts with a cross schema vertex field, the output will be a Field object
+      representing the property field used in the stitch
+      - If a property field of the expected name already exists in the selections one level
+        above (in parent_selections), the new Field will contain any existing directives of
+        this field
+    - Otherwise, the process will be repeated on child selections (if any) of the input AST
+      - If no child selection is modified, the exact same object as the input AST will be returned
+      - If some child selection is modified, a copy of the AST with the new selections will be
+        returned
+        - If a modified child selection is a property field
+          - If a property field of the same name already exists, the new field will replace the
+            existing field
```

Comment:
If I understand correctly, this could use the addition of `, while preserving any directives from the existing field`